File: android/src/main/java/au/com/crazybean/mobilex/kurir/modules/contacts/impl/ContactsModule.kt
package au.com.crazybean.mobilex.kurir.modules.contacts.impl

import au.com.crazybean.mobilex.kurir.modules.contacts.ContactsDelegate
import au.com.crazybean.mobilex.kurir.modules.contacts.ContactsView
import au.com.crazybean.mobilex.kurir.modules.contacts.ContactsViewModel
import org.koin.androidx.viewmodel.dsl.viewModel
import org.koin.dsl.module

val contactsModule = module {
    // ViewModel
    viewModel {
        ContactsViewModel(get(), get(), get())
    }

    // Delegate
    factory { (view: ContactsView?) ->
        ContactsDelegate(view, get())
    }
}

File: ios/Kurir/modules/contacts/impl/ContactsModule.swift
//
//  ContactsModule.swift
//  Kurir
//
//  Created by Loren on 19/9/19.
//  Copyright © 2019 Crazybean Studio. All rights reserved.
//

import Mobilex

class ContactsModule: Module {
    override func inject() {
        // ViewModel
        factory(ContactsViewModel.self) { r in
            ContactsViewModel(userData: r.resolve(UserData.self),
                              usersRepository: r.resolve(UsersRepository.self),
                              contactsRepository: r.resolve(ContactsRepository.self))
        }
        
        // Delegate
        factory(ContactsDelegate.self) { (r, view: ContactsView) -> ContactsDelegate in
            ContactsDelegate(view: view, viewModel: r.resolve(ContactsViewModel.self)!)
        }
    }
}
